crates/iota-genesis-builder/src/stardust/process_outputs.rs

```diff
@@ -78,110 +66,69 @@ struct OutputHeaderWithBalance {
     balance: u64,
 }
 
-// Define the trait for filtering logic
-trait HornetFilter<Item> {
-    /// Filters the given output based on custom criteria. It returns Some if
-    /// the output is modified in place or left as it is, but nonetheless
-    /// finalized. It returns None if the output requires other modifications
-    /// before being finalized (e.g., to be merged with other outputs).
-    fn process(&mut self, output: Item) -> Option<Item>;
-    /// Pop an output processed during the filtering process that was not
-    /// finalized.
-    fn pop_processed(&mut self) -> Option<Item>;
-    /// Print the metrics of the filtering process.
-    fn print_metrics(&self);
-}
-
 /// An iterator that processes outputs based on some filters.
-struct HornetFilterIterator<I, Item>
-where
-    I: Iterator<Item = Item>,
-{
-    /// The vector of filters.
-    filters: Vec<Box<dyn HornetFilter<Item>>>,
+struct HornetFilterIterator<I> {
     /// Iterator over `(OutputHeader, Output)` pairs.
     outputs: I,
+    /// Stores aggregated balances for eligible addresses.
```

Comment:
... I think in some way we need to make it simple to separate the scope of each filter. The original idea was to have a quick way to add or remove filters:
- e.g., imagine we need a new filter in a couple of sprints, with the original idea I only needed to create a new type implementing the trait.
- at the same tipe I can test out locally the different filters by simply modifying the `vec![scale_amount_filter, vesting_filter, voting_filter]` in `process_outputs_for_iota` or create dedicated `test_process_outputs` functions.